lib/pavilion/test_run.py

```diff
@@ -923,6 +929,80 @@ def create_id_dir(id_dir):
     def __repr__(self):
         return "TestRun({s.name}-{s.id})".format(s=self)
 
+    def _get_skipped(self):
+        match_list = []
+        match_list = self._match(match_list)
+
+        if len(match_list) is 0:
+            return False
+        else:
+            self.status.set(STATES.COMPLETE, match_list[0])
+            return True
+
+    def _make_regex(self, cond):
+        """_make_regex takes a dictionary and checks each value
+        under all keys and appends/prepends [^$] to make sure all
+        values are compatible with the regex match for conditional
+        statements. This is needed if the user is not using regex.
+        It explicitly defines the given value and only matches to it.
+        :param dict: Dictionary containing conditional statements
+        :return dict: Dictionary with updated values."""
+        cond_dict = self.config.get(cond, {})
+        for key in cond_dict:
+            list = []
+            for value in cond_dict[key]:
+                if len(value) is 0:
```

Comment:
`is` is not to be used when you mean `==`!

`a is b` means 'the pointer to a' == 'the pointer to b'. 

Are you absolutely sure there's only one pointer to the value 0?

Reply:
The answer is - it's implementation dependent. 

in cPython, yes there is only one value of zero (and most small integers), but not larger ones. (The limit is arbitrary and subject to change too.)
Where you see `is` a lot: `a is None`.

There is, according to the Python spec, only instance of None. 
In this case, `is` is both the proper and recommended comparison operator. 

It's otherwise used to check that two objects are the same object, as they might not be comparable by value.

Reply:
Good job checking for the empty string case though. I was impressed enough by that that I almost missed the whole `is` thing.